### ISSUE TITLE
How do you add storage keys to keyring storage?

### ISSUE THREAD

**Issue by whereisaaron** [gilbertchen/duplicacy#103]
You can use `-reset-passwords` to stop storing a storage key in a keyring, but how do you go the other way, start storing the storage key in a keyring?

Is there some way to used `duplicacy set -key XXX -value YYY` to enable key ring storage?

**Comment by gilbertchen** [comment]
This is currently not supported.  The closest I can think of is to run a command that is generally fast and has no side effect, like `list`.  You'll then get a change to enter the storage key or other passwords which will then be saved in the keyring.

**Comment by whereisaaron** [comment]
Thanks @gilbertchen I tried `list` but it still prompts every time. I tried `list` multiple times in a row, and it prompts for AWS key every time now. The keyring did work on the first run, but then I lost those setting and now I can't get it to use the keyring.

I ran with debug and I see errors "Failed to get the value from the keyring: dbus.Store: length mismatch"

```
Storage set to s3://us-east-1@amazon.com/my-bucket-name/foo
Reading the environment variable DUPLICACY_S3_ID
Failed to get the value from the keyring: dbus.Store: length mismatch
Enter S3 Access Key ID:AK1234567891
Reading the environment variable DUPLICACY_S3_SECRET
Failed to get the value from the keyring: dbus.Store: length mismatch
```

So I went to the keyring and deleted the three duplicacy entries I found there (password, s3 id, s3 key). I ran `list` again but I still get the same `dbus.Store: length mismatch` errors as above.

My preferences looks like:
```
[
    {
        "name": "default",
        "id": "Development",
        "storage": "s3://us-east-1@amazon.com/my-bucket-name/foo",
        "encrypted": true,
        "no_backup": false,
        "no_restore": false,
        "no_save_password": false,
        "keys": {}
    }
]
```

**Comment by gilbertchen** [comment]
This looks like a bug.  I'll take a look.

**Comment by gilbertchen** [comment]
Can you run dbus-monitor to see what calls it is making:
```
dbus-monitor interface=org.freedesktop.Secret.Service interface=org.freedesktop.Secret.Session interface=org.freedesktop.Secret.Collection interface=org.freedesktop.Secret.Item interface=org.freedesktop.Secret.Prompt
```

**Comment by niknah** [comment]
I am getting the same "Failed to get the value from the keyring: dbus.Store: length mismatch" error.

Here's the log from dbus-monitor


signal time=1504752909.063158 sender=org.freedesktop.DBus -> destination=:1.4825 serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
   string ":1.4825"
signal time=1504752909.063216 sender=org.freedesktop.DBus -> destination=:1.4825 serial=4 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.4825"
method call time=1504752912.001745 sender=:1.4826 -> destination=org.freedesktop.secrets serial=2 path=/org/freedesktop/secrets; interface=org.freedesktop.Secret.Service; member=OpenSession
   string "plain"
   variant       string ""
method call time=1504752912.002247 sender=:1.4826 -> destination=org.freedesktop.secrets serial=3 path=/org/freedesktop/secrets/session/s13; interface=org.freedesktop.Secret.Session; member=Close
method call time=1504752912.004216 sender=:1.4826 -> destination=org.freedesktop.secrets serial=4 path=/org/freedesktop/secrets; interface=org.freedesktop.Secret.Service; member=OpenSession
   string "plain"
   variant       string ""
method call time=1504752912.004613 sender=:1.4826 -> destination=org.freedesktop.secrets serial=5 path=/org/freedesktop/secrets; interface=org.freedesktop.Secret.Service; member=Unlock
   array [
      object path "/org/freedesktop/secrets/aliases/default"
   ]
method call time=1504752912.005058 sender=:1.4826 -> destination=org.freedesktop.secrets serial=6 path=/org/freedesktop/secrets/aliases/default; interface=org.freedesktop.Secret.Collection; member=SearchItems
   array [
      dict entry(
         string "username"
         string "password"
      )
      dict entry(
         string "service"
         string "duplicacy"
      )
   ]
method call time=1504752912.005511 sender=:1.4826 -> destination=org.freedesktop.secrets serial=7 path=/org/freedesktop/secrets/session/s14; interface=org.freedesktop.Secret.Session; member=Close

**Comment by gilbertchen** [comment]
If you can build from the source, changing [this line](https://github.com/gilbertchen/duplicacy/blob/197d20f0e07284c9c8a35d9d0800b366d53b2f18/src/duplicacy_keyring.go#L10) to import `github.com/tmc/keyring` instead of `github.com/gilbertchen/keyring` should fix it.

I made a [change](https://github.com/gilbertchen/keyring/commit/8273428a6ae58a7ecacddf911abcda82f927d02d) to that library because on my Ubuntu installation the method `org.freedesktop.Secret.Service.SearchItems` return two collections, one locked and the other unlocked.  However, it turns out this is [revision 1](https://specifications.freedesktop.org/secret-service/re01.html) of the specification; in [revision 2](https://specifications.freedesktop.org/secret-service/re02.html) the same method returns only one collection, thus the length mismatch error.

I'll fix it to handle both cases.

**Comment by niknah** [comment]
Thanks a lot!  It works now.  I just needed to create a password keyring called "login", maybe I deleted that by accident earlier.

**Comment by gilbertchen** [comment]
@niknah Did you mean that you fixed it by creating the "login" password keyring without the import change I suggested?

**Comment by niknah** [comment]
No, I recompiled it with your change.  It didn't work at first, until I added the "login" password keyring.

**Comment by gilbertchen** [comment]
This has been fixed by https://github.com/gilbertchen/keyring/commit/1260032d8c5455033c8aa592564d8ea0a6874935

You'll need to get that commit and build from source for the fix to work:

```
go get -u github.com/gilbertchen/keyrin
go build duplicacy/duplicacy_main.go
```

Let me know if you need a binary.
